common.md

```diff
@@ -0,0 +1,151 @@
+# Appendix: Common Varsig Headers
+
+Below are a few common signature headers and their fields. These are all given as the parts following the Varsig prefix and version.
+
+## RSA
+
+[RSASSA-PKCS #1 v1.5] signatures MUST include the following segments:
+
+``` abnf
+rsa-varsig = rsa-prefix rsa-hash-algorithm rsa-key-length
+rsa-prefix = %x1205 ; RSASSA-PKCS #1 v1.5
+rsa-hash-algorithm = unsigned-varint
+rsa-key-length = unsigned-varint
+```
+
+### Example: RS256
+
+| Segment              | Hexadecimal | Unsigned Varint | Comment                                 |
+|----------------------|-------------|-----------------|-----------------------------------------|
+| `rsa-prefix`     | `0x1205`    | `0x8524`        | RSASSA-PKCS #1 v1.5 [multicodec] prefix |
+| `rsa-hash-algorithm` | `0x12`      | `0x12`          | SHA2-256 [multicodec] prefix            |
+| `rsa-key-length`     | `varint`    | `varint`        | Length of public key in bytes           |
+
+### Example: RS512
+
+| Segment              | Hexadecimal | Unsigned Varint | Comment                                 |
+|----------------------|-------------|-----------------|-----------------------------------------|
+| `rsa-prefix`     | `0x1205`    | `0x8524`        | RSASSA-PKCS #1 v1.5 [multicodec] prefix |
+| `rsa-hash-algorithm` | `0x13`      | `0x13`          | SHA2-512 [multicodec] prefix            |
+| `rsa-key-length`     | `varint`    | `varint`        | Length of public key in bytes           |
+
+## [EdDSA]
+
+``` abnf
+eddsa-varsig = eddsa-prefix eddsa-curve eddsa-hash-algorithm
+eddsa-prefix = %xED
+eddsa-curve = unsigned-varint
+eddsa-hash-algorithm = unsigned-varint
+```
+
+### Example: Ed25519
+
+| Segment                | Hexadecimal | Unsigned Varint | Comment                          |
+|------------------------|-------------|-----------------|----------------------------------|
+| `eddsa-prefix`         | `0xED`      | `0xED01`        | EdDSA prefix                     |
```

Comment:
There's a draft `multicodec.Code` for EdDSA that's `0xd0ed` and is labeled that it was created for `varsig`, should that be used here since `0xed` is specifically the `Ed25519` curve (technically the pub key)

Reply:
When working with Varsig, you can safely ignore Multicodec entries.

Varsig sometimes puns on Multicodec values, but they're distinct values. Varsig's only relationship to multicodec is that Varsig itself has an entry in Multicodec. Multicodec is very difficult to get changes into (sometime 6+ months lag time), so we can add stuff to this spec much more easily.